config/src/main/java/org/springframework/security/config/annotation/web/builders/HttpSecurity.java

```diff
@@ -1094,6 +1095,42 @@ private ApplicationContext getContext() {
 		return getOrApply(new HttpBasicConfigurer<>());
 	}
 
+	/**
+	 * Configures HTTP Basic authentication.
+	 *
+	 * <h2>Example Configuration</h2>
+	 *
+	 * The example below demonstrates how to configure HTTP Basic authentication for an
+	 * application. The default realm is "Realm", but can be
+	 * customized using {@link HttpBasicConfigurer#realmName(String)}.
+	 *
+	 * <pre>
+	 * &#064;Configuration
+	 * &#064;EnableWebSecurity
+	 * public class HttpBasicSecurityConfig extends WebSecurityConfigurerAdapter {
+	 *
+	 * 	&#064;Override
+	 * 	protected void configure(HttpSecurity http) throws Exception {
+	 * 		http.authorizeRequests().antMatchers(&quot;/**&quot;).hasRole(&quot;USER&quot;).and().httpBasic(httpBasicConfig -> {});
```

Comment:
I'd fix this formatting to have newlines in it. I know there are other places we are missing newlines, but that occurred by mistake.